testing/test_states.py

```diff
@@ -772,3 +780,212 @@ def test_graph_masks_device_on_cuda(datas):
     assert stacked.device.type == "cuda"
     _assert_tensordict_on_device(stacked.forward_masks, stacked.device)
     _assert_tensordict_on_device(stacked.backward_masks, stacked.device)
+
+
+def _cpu_devices():
+    return torch.device("cpu"), torch.device("cpu")
+
+
+def _cpu_and_meta_devices():
+    # Meta is a dummy device for mismatch-only tests; do not transfer back from meta
+    return torch.device("cpu"), torch.device("meta")
```

Comment:
I think removing these functions (and write the one line implementation instead) will make tests clearer

Reply:
makes sense!